src/mon/OSDMonitor.cc

```diff
@@ -6052,6 +6052,10 @@ int OSDMonitor::prepare_command_pool_set(const cmdmap_t& cmdmap,
          << " (you may adjust 'mon max pool pg num' for higher values)";
       return -ERANGE;
     }
+    if (creating_pgs.is_creating_pool(pool)) {
+      ss << "still creating initial PGs; cannot update pg_num yet";
+      return -EBUSY;
+    }
```

Comment:
...and actually, I think we *do* handle splits properly now, so this patch can get pulled out?

Reply:
The problem, IIRC, was that a PG split target might already exist on the OSD.  This could happen if pg 1.0 splits to 1.2 on osd.0, 1.2 peers with osd.1 to notify/create the pg, and osd.1 only then processes the 1.0 -> 1.1 split (the split is async from the map processing).  Something like that.

It was a hard race to hit in QA but I couldn't come up with a clean way to avoid it before realizing that splitting while a pool was still being created wasn't a necessary (or particularly useful) requirement.

Reply:
So what does happen if somebody runs into that race? Are we okay with just a release note telling people not to do that?

Reply:
I am!

Reply:
Sorry, I mean:
If this happens, does the child PG not get created and IO to it blocks? Does the OSD crash and everything gets fixed up on reboot? Is the cluster permanently broken?

Without looking at it seriously I'd assume it just hangs but can be fixed by rebooting some set of involved OSDs, in which case a release note sounds fine. But if we're talking about data destruction I think we need something more serious.

Reply:
The pg_num change wouldn't happen, and the user would see something like
<pre>
# ceph osd pool create foo 1024 ; ceph osd pool set foo 2048
error EBUSY: still creating initial PGs; cannot update pg_num yet
</pre>

Reply:
I mean if they upgrade osds before monitors, and this code isn't present for them.

Reply:
Oh I see. I think it will either be a osd crash + restart, or (worst case) a ceph-objectstore-tool invocation to remove the empty child so that the split can proceed.

I suggest backporting the above check to luminous to reduce the risk.